lib/websocket_adapter.rb

```diff
@@ -0,0 +1,9 @@
+module WebsocketAdapter
+  def self.new(record, env, sock)
```

Comment:
this feels funky.

At first I felt that you should pass protocol to the parent. (not define the method, but rather pass to the parent in the initializer)

Now it feels like there should be 2 different parent classes here. - deriving form Legacy or Standard.

Overriding `new` is always sketchy

Reply:
It's not an override, it's a `Module`

Reply:
Also it doesn't really make sense with 2 items, but `Purr` could be plugged in here very easily if it's done.